src/main/java/com/google/sps/model/AuthenticationVerifierImpl.java

```diff
@@ -41,6 +43,8 @@ public boolean verifyUserToken(String idToken) throws GeneralSecurityException,
     // If the userToken is not null, the identity is verified and vice versa
     GoogleIdToken userToken = verifier.verify(idToken);
 
-    return userToken != null;
+    return userToken != null
+        ? Optional.ofNullable(userToken.getPayload().getEmail())
+        : Optional.empty();
```

Comment:
There is an even cleaner way to do this with optionals:
```java
return Optional.ofNullable(userToken).map(t -> t.getPayload().getEmail())
```

Reply:
Nifty! Fixed